packages/wallet-adapter-core/src/conversion.ts

```diff
@@ -0,0 +1,101 @@
+import { NetworkName } from "./constants";
+
+
+// we could serialize the entire thing and then deserialize
+// export const convertEntryFunction(entryFunction: EntryFunction): TxnBuilderTypes.EntryFunction {
+//     const newModuleName = entryFunction.module_name;
+//     const newFunctionName = entryFunction.function_name;
+//     const newArgs = entryFunction.args;
+//     const newTypeArgs = entryFunction.type_args;
+
+//     const oldHexAddress = HexString.fromUint8Array( newModuleName.address.data );
+//     const oldAddress = TxnBuilderTypes.AccountAddress.fromHex(oldHexAddress);
+//     const oldModuleIdentifier = new TxnBuilderTypes.Identifier(newModuleName.name.identifier);
+//     const oldModuleName = new TxnBuilderTypes.ModuleId(oldAddress, oldModuleIdentifier);
+//     const oldFunctionName = new TxnBuilderTypes.Identifier(newFunctionName.identifier);
+
+//     const oldArgs = newArgs.map((arg) => { arg.bcsToBytes() });
+//     const oldTypeArgs = Array<TxnBuilderTypes.TypeTag>();
+
+//     return new TxnBuilderTypes.EntryFunction(
+//         oldModuleName,
+//         oldFunctionName,
+//         oldTypeArgs,
+//         oldArgs,
+//     )
+// }
+
+// old => new
+export function convertNetwork(networkInfo: NetworkInfo | null): Network {
+    switch(networkInfo?.name) {
+        case "mainnet" as NetworkName:
+            return Network.MAINNET;
+        case "testnet" as NetworkName:
+            return Network.TESTNET;
+        case "devnet" as NetworkName:
+            return Network.DEVNET;
+        default:
+            throw new Error("Invalid network name")
+    }
+}
+
+// new => old
+export function convertToBCSPayload(payload: TransactionBuilderTypes.TransactionPayload): TxnBuilderTypes.TransactionPayload {
+    const deserializer = new BCS.Deserializer(payload.bcsToBytes());
+    return TxnBuilderTypes.TransactionPayload.deserialize(deserializer)
+}
+declare type TransactionPayload = TransactionPayloadEntryFunction | TransactionPayloadScript | TransactionPayloadMultisig;
+
+/*
+// new => old
+export function convertToJSONPayload(payload: TransactionPayload): Types.TransactionPayload {
+    let payloadType: string;
+    let bytecode: Uint8Array;
+    let type_arguments: string;
+    let function_args: any;
+    if (payload instanceof TransactionPayloadEntryFunction) {
+        payloadType = "entry_function_payload"
+        type_arguments = payload.entryFunction.type_args; // need a .toString() function for this
+        function_args = payload.entryFunction.args.map((arg) => { arg.value }); // would need `inner` or `getRawValues` in EntryFunctionArgumentTypes
+    } else if (payload instanceof TransactionPayloadScript) {
+        payloadType = "script_payload"
+    } else if (payload instanceof TransactionPayloadMultisig) {
+        payloadType = "multisig"
+        code = 
+    }
+}
+*/
+
+// new => old
+export function convertAuthenticator(authenticator: AccountAuthenticator): TxnBuilderTypes.AccountAuthenticator {
```

Comment:
looks like we dont really use this and some other functions in this file, do we need to keep them part of this PR?